s3v2/s3.go

```diff
@@ -395,8 +395,8 @@ func (s *S3File) openRead() error {
 
 	s.lock.Lock()
 	s.readOpened = true
-	if hoo.ContentLength != 0 {
-		s.fileSize = hoo.ContentLength
+	if hoo.ContentLength != nil {
```

Comment:
```suggestion
	if hoo.ContentLength != nil && *hoo.ContentLength != 0 {
```
I think just checking if it's not nil is not gonna provide exactly the same behavior as the current version